### PR TITLE
add readme and docs

### PR DESCRIPTION
**Pull request by tanbowensg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,46 +11,85 @@
   </a>
   <img alt="Github Stars" src="https://badgen.net/github/stars/webzard-io/sunmao-ui" />
 </p>
+Sunmao 是一个前端低代码框架。通过Sunmao，您可以轻松将各种前端UI组件库自己开发的前端组件，封装成低代码组件库，从而搭建您自己的低代码UI开发工具，使前端开发变得如榫卯般严丝合缝。
```
Comment (MrWindlike) [inline README.md:14]:
```diff
- 您可以轻松将各种前端UI组件库自己开发的前端组件
+ 您可以轻松将各种前端UI组件库和自己开发的前端组件
```

File: README.md [hunk 2]
```diff
@@ -10,46 +11,85 @@
...
+
+开发者的职责和寻常的前端开发者类似，负责开发 UI 组件，并且把普通的 UI 组件封装为 Sunmao 的组件。开发者需要通过写代码来实现组件的逻辑。
+
+使用者的职责是利用开发者封装好的 Sunmao 组件，在 Sunmao 低代码编辑器中搭建前端应用。使用者不需要前端知识和编程能力，仅通过 UI 交互就可以完成应用搭建。
+
+为此，我们准备了两份教程，分别面向不同的角色。使用者仅需阅读使用者的教程，但开发者则要先阅读使用者教程，再阅读开发者教程。
+
+* [使用者的教程](./user.md)
+* [开发者的教程](./developer.md)
+
```
Comment (xzdry) [inline README.md:86]:
```diff
- * [使用者的教程](./user.md)
- * [开发者的教程](./developer.md)
+ * [开发者的教程](./docs/zh/user.md)
+ * [开发者的教程](./docs/zh/developer.md)
```

File: docs/zh/component.md [hunk 3]
```diff
@@ -0,0 +1,505 @@
+# Component 开发文档
+
+Component由两部分构成，一个是Spec，一个是Implementation。如果把Component想象成一个类的话，Spec就是类的接口，Implementation就是类的实现。
```
[the remark this answers is not in the document]
Comment (Yuyz0112) [reply docs/zh/component.md:3]: prettier can format markdown like this

File: docs/zh/component.md [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+    mergeState({
+      value,
+    });
+  }, [mergeState, value]);
+
+  return <Input value={value} onChange={(newVal) => setValue(newVal)} />;
+};
+```
+
+> 如果是使用 React 开发的 Component，建议把 mergeState 放到 useEffect 中执行。
```
[the remark this answers is not in the document]
Comment (Yuyz0112) [reply docs/zh/component.md:202]: +1

File: docs/zh/component.md [hunk 5]
```diff
@@ -0,0 +1,517 @@
+# Component 开发文档
+
+Componen t 由两部分构成，一个是 Spec，一个是 Implementation。如果把 Component 想象成一个类的话，Spec 就是类的接口，Implementation 就是类的实现。
```
Comment (MrWindlike) [inline docs/zh/component.md:3]: `Componen t` -> `Component`

**Review by MrWindlike — APPROVED** [review]

**Pull request merged**